### PR TITLE
Adds BuildImageSteps to encompass the entire image building process

### LINKED ISSUE #117: Each layer should have independent parallelizable build step.
Rather than be built/pushed sequentially.

The build step should also check for existence before pushing.

### LINKED ISSUE #119: Check if layer needs rebuilding in BuildAndCacheApplicationLayersStep.

### PR DESCRIPTION
**Pull request by coollog**
Requires #112 

Fixes #119 
Fixes #117 

Adds a `BuildImageSteps` that can run all the steps for building an entire image. These steps were previously in `StepIntegrationTest`.

### CODE REVIEW

File: crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java [hunk 1]
```diff
@@ -16,17 +16,19 @@
...
 import java.io.UncheckedIOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.attribute.FileTime;
 import java.time.Instant;
+import java.util.List;
 import java.util.Optional;
-import java.util.Set;
 
+// TODO: Put into CacheReader and remove this class.
```
Comment (coollog) [inline crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java]: Filed https://github.com/GoogleCloudPlatform/minikube-build-tools-for-java/issues/121

File: crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java [hunk 2]
```diff
@@ -27,6 +28,7 @@
 import java.util.List;
 import java.util.Optional;
 
+// TODO: Put into CacheReader and remove this class.
```
Comment (coollog) [inline crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java]: Filed https://github.com/GoogleCloudPlatform/minikube-build-tools-for-java/issues/121

File: crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java [hunk 3]
```diff
@@ -74,26 +76,45 @@ public CacheChecker(Cache cache) {
...
+          <= 0) {
+        // This layer is an up-to-date layer.
+        return cachedLayer;
+      }
+    }
+
+    return null;
   }
 
+  // TODO: REMOVE
```
Comment (coollog) [inline crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java]: #121 will include removing this method, which isn't used anymore.

File: crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java [hunk 4]
```diff
@@ -74,26 +76,45 @@ public CacheChecker(Cache cache) {
...
   }
 
-  /** @return true if the base image layers exist in the cache; false otherwise */
-  public boolean areBaseImageLayersCached(ImageLayers<ReferenceLayer> baseImageLayers)
-      throws CacheMetadataCorruptedException {
-    // Gets the base image layers.
-    ImageLayers<CachedLayerWithMetadata> cachedLayers =
-        cache.getMetadata().filterLayers().byType(CachedLayerType.BASE).filter();
+  /**
+   * @return the cached layer with digest {@code layerDigest} exists in the cache; {@code false}
```
Comment (loosebazooka) [inline crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java]: update this comment?

**Pull request merged**